bundles/org.eclipse.core.resources/src/org/eclipse/core/resources/ResourcesPlugin.java

```diff
@@ -416,12 +417,16 @@ public static ResourcesPlugin getPlugin() {
 	 * Returns the workspace. The workspace is not accessible after the resources
 	 * plug-in has shutdown.
 	 *
-	 * @return the workspace that was created by the single instance of this
-	 *   plug-in class.
+	 * @deprecated Accessing the Workspace in a static way is prone to start-up
+	 *             order problem, please consider using any of your favorite
+	 *             techniques (Declarative Services, ServiceTracker, Blueprint, ...)
+	 * @return the workspace that was created by the single instance of this plug-in
+	 *         class.
 	 */
+	@Deprecated(forRemoval = true)
```

Comment:
In case of removal it should be a major version change, perhaps for the whole Eclipse SDK, since the most  part of Eclipse plug-ins that use IResource will stop work.

Reply:
Yep, that's why I just marked it for removal so if we ever do a major upgrade...

Reply:
Please change to "forRemoval = false" and that could be discussed (after fixing all the new warnings in resources code and of others agree).

"forRemoval = true" is not an option, sorry, see comments from Ed and others.

Reply:
I don't get why this makes any difference here but anyways I'll remove the "offending" flag then, this literally makes no sense to me as a major version change would require all plugins to become invalid anyways...